### PR TITLE
Allows for more logging levels

### PR DESCRIPTION
**Pull request by rjfonseca**
We were parsing less logging levels that the zerolog package provides.
This is an older code and zerolog supports more levels now and has a
parsing functions.

Change how we parse levels to use zerolog's parsing function so we
support the same logging levels that they do.

Also adds trace level logging for the app package, so it's easier to
understand what the app package is doing through the logs.

### CODE REVIEW

**Review by victormn — APPROVED** [review]

File: app/app.go [hunk 1]
```diff
@@ -115,14 +117,18 @@ func MustNew(ctx context.Context, adminPort string) *App {
 }
 
 // Shutdown calls all shutdown methods, in order they were added.
-func (a *App) Shutdown(ctx context.Context) error {
+func (a *App) Shutdown(ctx context.Context) (err error) {
+	log.Trace().Msg("[app] Starting graceful shutdown")
+	defer log.Trace().Err(err).Msg("[app] Graceful shutdown finished")
```
Comment (eric-reis) [inline app/app.go:122]: Nit: I'm not sure, but I think that `err` is evaluated now, not when executing the defer function.
https://play.golang.org/p/Q0QbyeGcTmv

**Review by eric-reis — APPROVED** [review]
Approving, but indicated a potential minor bug

File: app/app.go (hunk 1 above)
Comment (rjfonseca) [reply app/app.go]: You are totally right. I've fixed this issue in a new commit and added a couple of more messages for make it easier debug the app livecycle.

**Review by eric-reis — APPROVED** [review]
I like the new changes!

**Pull request merged**